protodroid-no-op/src/main/java/id/lukasdylan/grpc/protodroid/SingletonHolder.kt

```diff
@@ -0,0 +1,29 @@
+package id.lukasdylan.grpc.protodroid
+
+@Suppress("unused")
+open class SingletonHolder<out T : Any, in A>(creator: (A) -> T) {
```

Comment:
can this class mark as internal class?

Reply:
Since `Protodroid` companion extends this, it exposes it's internal methods `getInstance` etc, so will need to be open, unless you don't want to expose `Protodroid.getInstance(context)`?

Reply:
Yes i want to hide implementation of this class since it contain logic on internal repositories that should not accessed outside library

Reply:
So to clarify, you aren't wanting Protodroid to be public? The only methods a consumer could use would be the one's that are public in `Protodroid`
<img width="827" alt="Screen Shot 2021-09-20 at 5 56 50 pm" src="https://user-images.githubusercontent.com/88014909/133961773-f6d96c33-d045-46b0-bc71-79a14adb80ae.png">
c